hazelcast-sql/src/main/java/com/hazelcast/sql/impl/FieldsUtil.java

```diff
@@ -83,6 +88,22 @@ public static SortedMap<String, Class<?>> resolveClass(@Nonnull Class<?> clazz)
         return fields;
     }
 
+    /**
+     * Constructs instance-to-name mapping for enum-like classes.
+     */
+    @SuppressWarnings("unchecked")
+    public static <T> Map<T, String> getEnumConstants(Class<T> type) {
+        return Arrays.stream(QueryDataType.class.getDeclaredFields())
+                .filter(field -> field.getType() == type && field.getModifiers() == (PUBLIC | STATIC | FINAL))
```

Comment:
you should not compare flags using `==` but with mask. `==` will work as long as no new flag is used in modifiers

Reply:
Moved to the upcoming PR.